### PR TITLE
Add xcm tests

### PR DESCRIPTION
**Pull request by N1ghtStorm**

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,8 @@ targets = ["x86_64-unknown-linux-gnu"]
 substrate-wasm-builder = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.31" }
 
 [dependencies]
-hex-literal = { version = '0.3.1', optional = true }
+# hex-literal = { version = '0.3.1', optional = true }
```
Comment (wer1st) [inline runtime/Cargo.toml]: Why do we need commented line?

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -171,7 +183,7 @@ std = [
 ]
 
 runtime-benchmarks = [
-	"hex-literal",
+	# "hex-literal",
```
Comment (wer1st) [inline runtime/Cargo.toml]: The same question as above

File: runtime/src/xcm_tests/mod.rs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	.unwrap();
+
+	let mut ext = TestExternalities::new(t);
+	ext.execute_with(|| System::set_block_number(1));
+	ext
+}
+
+pub struct CurrencyIdConvert;
+impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
```
Comment (wer1st) [inline runtime/src/xcm_tests/mod.rs:197]: Indents look too big

File: runtime/src/xcm_tests/mod.rs [hunk 4]
```diff
@@ -0,0 +1,203 @@
+pub mod para_x;
```
Comment (wer1st) [inline runtime/src/xcm_tests/mod.rs:31]: Copyright header

File: runtime/src/xcm_tests/para_x.rs [hunk 5]
```diff
@@ -0,0 +1,282 @@
+use super::{Amount, Balance, CurrencyId, CurrencyIdConvert, ParachainXcmRouter};
```
Comment (wer1st) [inline runtime/src/xcm_tests/para_x.rs]: Copyright header

File: runtime/src/xcm_tests/relay.rs [hunk 6]
```diff
@@ -0,0 +1,161 @@
+use frame_support::{
```
Comment (wer1st) [inline runtime/src/xcm_tests/relay.rs:31]: Header

File: Cargo.lock [hunk 7]
```diff
@@ -11707,7 +11740,7 @@ version = "1.6.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "97fee6b57c6a41524a810daee9286c02d7752c4253064d0b05472833a438f675"
 dependencies = [
- "cfg-if 1.0.0",
+ "cfg-if 0.1.10",
```
Comment (wer1st) [inline Cargo.lock]: Why more old version?

File: runtime/src/xcm_tests/tests.rs [hunk 8]
```diff
@@ -0,0 +1,466 @@
+#![cfg(test)]
```
Comment (wer1st) [inline runtime/src/xcm_tests/tests.rs]: Header

File: runtime/src/xcm_tests/mod.rs (hunk 3 above)
Comment (wer1st) [reply runtime/src/xcm_tests/mod.rs:197]: Decided to fix it in separate PR

Comment (N1ghtStorm) [reply runtime/src/xcm_tests/mod.rs:197]: created task for that https://github.com/sora-xor/sora2-parachain/issues/74

File: runtime/src/xcm_tests/tests.rs [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+
+fn para_x_asset_id() -> crate::H256 {
+	hex_literal::hex!("54fd1e1728cd833d21da6f3e36c50884062e35edfc24aec7a70c18a60451255a").into()
+}
+
+fn message_id() -> crate::H256 {
+	hex_literal::hex!("54fd1e1728cd833d21da6f3e36c50884062e35edfc24aec7a70c18a60451255c").into()
+}
+
+fn prepeare_sora_parachain() {
```
Comment (vovac12) [inline runtime/src/xcm_tests/tests.rs]:
```suggestion
fn prepare_sora_parachain() {
```

**Review by vovac12 — APPROVED** [review]

**Pull request merged**